src/generators/mappersGenerator.ts

```diff
@@ -21,29 +28,132 @@ export function generateMappers(
     moduleSpecifier: "@azure/core-http"
   });
 
-  for (const mapper of clientDetails.mappers) {
-    // Generate the mapper definition and replace object key strings
-    // with the unquoted name
-    const mapperString = JSON.stringify(mapper, undefined, 2).replace(
-      /\"([^"]+)\":/g,
-      "$1:"
-    );
+  let discriminators: { [key: string]: string } = {};
 
+  for (const mapper of clientDetails.mappers) {
     if (mapper.type.name === MapperType.Composite && mapper.serializedName) {
+      const compositeMapper = mapper as CompositeMapper;
+
+      if (compositeMapper.type.polymorphicDiscriminator) {
+        const { uberParent, className } = compositeMapper.type;
+        if (!uberParent || !className) {
+          throw new Error(
+            `Expected CompositeMapper with polymorphicDiscriminator to specify uberParent property uberParent: ${uberParent} className: ${className}`
```

Comment:
Not super important, just adding parentheses to make the error message a little clearer
```suggestion
            `Expected CompositeMapper with polymorphicDiscriminator to specify uberParent property (uberParent: ${uberParent}, className: ${className})`
```